spec/controllers/books_controller_spec.rb

```diff
@@ -5,81 +5,159 @@
+      before { sign_in user }
+
+      context 'when validation success' do
+        it 'creates a new book' do
+          expect { subject }.to change(Book, :count).by(1)
+          expect(subject).to redirect_to(Book.last)
+          expect(subject.status).to eq(302)
+        end
+      end
+
+      context 'when validation fail' do
+        let(:params) { { book: { name: '', description: '', release: '' }, format: :json } }
+
+        it 'raise RecordInvalid  if no book detail' do
+          expect(subject.status).to eq(422)
+          expect(JSON.parse(response.body)['error']).to include('Validation failed')
+        end
       end
     end
 
-    context 'when validation fail' do
-      let(:params) { { book: { name: '', description: '', release: '' }, format: :json } }
-      it 'raise RecordInvalid  if no book detail' do
-        expect(subject.status).to eq(422)
-        expect(JSON.parse(response.body)["error"]).to include("Validation failed")
+    context 'when user not signin' do
+      it 'return status 302' do
+        expect(subject.status).to eq(302)
+        expect(subject).to redirect_to(new_user_session_path)
       end
     end
   end
 
   describe 'PUT #update' do
     subject { put :update, params: { id: book.id, book: book_attr }, format: :json }
 
-    let!(:book) { create(:book) }
-    context 'when validation success' do
-      let(:book_attr) { attributes_for(:book) }
-      it 'Update the book' do
-        expect(subject).to redirect_to(Book.find(book.id))
-        expect(book.name).to_not eq(Book.find(book.id).name)
+    let!(:user) { create(:user) }
+    let!(:book) { create(:book, user_id: user.id) }
+
+    context 'Authorized user' do
+      before { sign_in user }
+
+      context 'when validation success' do
+        let(:book_attr) { attributes_for(:book).merge(user_id: user.id) }
+
+        it 'Update the book if user authorized' do
+          expect(subject).to redirect_to(Book.find(book.id))
+          expect(book.name).to_not eq(Book.find(book.id).name)
+        end
+      end
+
+      context 'when validation fail' do
+        let(:book_attr) { attributes_for(:book, name: '').merge(user_id: user.id) }
+
+        it 'response validation fail' do
+          expect(subject.status).to eq(422)
+          expect(JSON.parse(response.body)['error']).to include('Validation failed')
+        end
       end
     end
 
-    context 'when validation fail' do
-      let(:book_attr) { attributes_for(:book, name: '') }
-      it 'raise RecordInvalid if no book detail' do
-        expect(subject.status).to eq(422)
-        expect(JSON.parse(response.body)["error"]).to include("Validation failed")
+    context 'Unauthorized user' do
+      let(:book_attr) { attributes_for(:book).merge(user_id: user.id) }
+      let(:other_user) { create(:user) }
+
+      before { sign_in other_user }
+
+      it 'Update fail if user now authorized' do
+        expect { subject }.to raise_error(Pundit::NotAuthorizedError)
```

Comment:
I think if user has no authorization, we should return `forbidden(403)` back instead of raising the error